src/concrete_convex_hull.jl

```diff
@@ -99,27 +99,10 @@ function convex_hull!(points::Vector{VN};
 
     n = length(first(points))
 
-    # two points
-    if m == 2
-        if n == 1
-            p1, p2 = points[1], points[2]
-            if p1 == p2  # check for redundancy
-                pop!(points)
-            elseif p1[1] > p2[1]
-                points[1], points[2] = p2, p1
-            end
-        elseif n == 2
-            # special case, see #876
-            p1, p2 = points[1], points[2]
-            if p1 == p2  # check for redundancy
-                pop!(points)
-            elseif p1 <= p2
-                nothing
-            else
-                points[1], points[2] = p2, p1
-            end
-        end
-        return points
+    if m == 2 # two points
+        return _two_points_2d!(points)
+    elseif m == 3 # three points
+        return _three_points_2d!(points)
```

Comment:
This is switch block is currently not correct (you call the functions with arbitrary dimension). I would suggest to move up the dimension checks (on `n`) and integrate this switch on `m` in the `n == 2` case.